src/Compilers/CSharp/Portable/Symbols/AbstractTypeMap.cs

```diff
@@ -47,6 +47,15 @@ internal NamedTypeSymbol SubstituteNamedType(NamedTypeSymbol previous)
                 return (oldFieldTypes == newFieldTypes) ? previous : AnonymousTypeManager.ConstructAnonymousTypeSymbol(previous, newFieldTypes);
             }
 
+            if (previous.IsTupleType)
+            {
+                TupleTypeSymbol previousTuple = (TupleTypeSymbol)previous;
+                NamedTypeSymbol oldUnderlyingType = previousTuple.UnderlyingTupleType;
+                NamedTypeSymbol newUnderlyingType = (NamedTypeSymbol)SubstituteType(previousTuple.UnderlyingTupleType).Type;
```

Comment:
right. I considered removing the temp here, but i need it anyways, so why not to use it here.